### PR TITLE
Improve webserver setup instructions

### PR DESCRIPTION
**Pull request by gggeek**
english improvements; mention ExecCGI for Apache; move nginx instructions to .md format; format Varnish instructions same as other ones

### CODE REVIEW

**Comment by gggeek** [comment]
PS: One thing which I feel is currently under documented is the creation of the database schema.
I think it could be detailed in install.md as a dedicated step - without going as far as including the sql for a given type of db, but mentioning that the character set has to be utf8.
The reasoning is that:
- currently it is easy to miss in the docs the mention that the db has to be created already
- it is more or less the only manual step which is not documented, if we assume that the user starts with a plain lamp stack

As an alternative (or as addition): put the sql for db schema creation in doc/, next to the vhost examples...

**Comment by yannickroger** [comment]
Regarding the nginx documentation being in RST, it was because we were planning on using readthedocs at some point and it would only support RST.
ping @mere-teresa  @andrerom

File: doc/apache2/Readme.md [hunk 1]
```diff
@@ -36,12 +36,12 @@ Configuration
 
 #### Pure legacy setup
 
-In eZ Publish 5.x you can optionally set it up to only use legacy, this is useful for 4.x upgrades which only uses legacy.
-*Warning: By doing this, absolutely no integrations between legacy and Platform works, so you can not use any of the Platform features (API, HttpCache, Symfony, ..) in this setup.*
+eZ Publish 5.x can optionally be set up to only use the legacy kernel. This is useful for 4.x upgrades which only use legacy.
```
Comment (yannickroger) [inline doc/apache2/Readme.md]: Should we also rename eZ Publish 5.x to eZ Platform ?

**Comment by yannickroger** [comment]
+1 (with my comments sorted out ;) )

File: doc/apache2/vhost.template [hunk 2]
```diff
@@ -14,12 +14,13 @@
     # LogLevel debug
 
     <Directory %BASEDIR%/web>
+        # If using php configured in FastCGI mode, you will also need to add "ExecCGI" to the line below
```
Comment (andrerom) [inline doc/apache2/vhost.template]: do we? this is used directly for Behat tests on travis, and seems to work w/o there.

**Comment by andrerom** [comment]
@gggeek would prefer if it is kept as rst for now, and that we do that choice separately, it also possible affect links.

File: doc/apache2/vhost.template (hunk 2 above)
Comment (gggeek) [inline doc/apache2/vhost.template]: It did not work without it on my setup: apache 2.4 + mod_fcgid. I did not test using plain mod_fastcgi, but according to google it does need it as well (at least it says so in the Arch Linux wiki).
Are you sure about the Travis Apache config?

**Comment by gggeek** [comment]
About rst: I'd personally prefer to change it now (for uniformity) and change it back when/if all of the doc changes format. But I see your point about links. Undoing

File: doc/apache2/vhost.template (hunk 2 above)
Comment (andrerom) [inline doc/apache2/vhost.template]: It's probably 2.2 and not 2.4 as travis uses old version of Ubuntu since they are afraid of breaking stuff for people.

you can see what is used here:
- https://github.com/ezsystems/ezplatform/blob/master/bin/.travis/configure_apache2.sh
- https://github.com/ezsystems/ezplatform/blob/master/bin/.travis/apache2/php5-fcgi

**Comment by gggeek** [comment]
While at it, renamed eZ Publish to eZ Platform in all doc occurrences

File: doc/apache2/vhost.template [hunk 3]
```diff
@@ -164,14 +162,14 @@
...
+        <LocationMatch "^/share/icons/.*">
+            # Icons as used by admin interface, barely change
             ExpiresActive on
             ExpiresDefault "now plus 7 days"
         </LocationMatch>
 
-        <LocationMatch "^/share/icons/.*">
-            # Icons as used by admin interface, barly change
+        <LocationMatch "^/design/[^/]+/(stylesheets|images|javascripts?|lib|flash)/.*">
+            # Same as above for elements bundled eZ Publish designs
```
Comment (andrerom) [inline doc/apache2/vhost.template]: could actually remove all legacy stuff here

however you are starting to do to many things in same PR as usual, so if further legacy removal stuff is found, do it in separate PR.

**Comment by gggeek** [comment]
PS: changed the comment mentioning need for ExecCgi to 'might need'. I can not test on windows using fastcgi as the dlls have vanished off the net, and do not have fcgid.dll for apache 2.2, either.

Will squash the commits when we agree that all is good.

File: doc/apache2/vhost.template (hunk 3 above)
Comment (gggeek) [inline doc/apache2/vhost.template]: It was in my todo list to test existance of legacy dirs.

Any idea about how to split this PR in a meaningful way?

Comment (andrerom) [inline doc/apache2/vhost.template]: followup

**Comment by andrerom** [comment]
+1, sorry, should have pinged you for squash, but ok.

File: doc/apache2/vhost.template (hunk 3 above)
Comment (gggeek) [inline doc/apache2/vhost.template]: will do
